src/lightning/pytorch/core/saving.py

```diff
@@ -59,6 +59,8 @@ def _load_from_checkpoint(
     **kwargs: Any,
 ) -> Union["pl.LightningModule", "pl.LightningDataModule"]:
     with pl_legacy_patch():
+        if map_location is None:
+            map_location = torch.device("cuda") if not torch.cuda.is_available() else None
```

Comment:
I think this would be more like what we wanted to:
```suggestion
        map_location = "cpu" if not torch.cuda.device_count() else None
```
This says that if there are no GPUs, set map_location="cpu" just to be save in case the checkpoint contains cuda tensors. This is what we want I think.

The previous logic you had would have set map_location to a CUDA device when cuda was not available, which wouldn't work well.

Reply:
My second recommendation on top of this would be to use our CUDAAccelerator class:

```suggestion
        from lightning.pytorch.accelerators import CUDAAccelerator

        map_location = "cpu" if not CUDAAccelerator.is_available() else None
```

Reply:
LGTM, I am sorry for the weird PR suggestion (Somehow, I tested it on my installed version and committed it in the pulled repo (instead of copying, I rewrote it and had a brain freeze).

Reply:
WDYT about importing the `CUDAAccelerator` at the top of the file? 
Furthermore, I would not override the `map_location` (wrt. backwards compatibility) if the user wants to pass an argument him/herself?